eternalcore-core/src/main/java/com/eternalcode/core/feature/jail/JailServiceImpl.java

```diff
@@ -0,0 +1,331 @@
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.entity.Player;
+
+import javax.annotation.Nullable;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Service
+public class JailServiceImpl implements JailService {
+
+    private final TeleportService teleportService;
+    private final SpawnService spawnService;
+    private final NoticeService noticeService;
+    private final JailSettings settings;
+    private Position jailPosition;
+    private final JailLocationRepository jailRepository;
+
+    private final PrisonersRepository prisonersRepository;
+
+
+    private final Map<UUID, Prisoner> jailedPlayers = new ConcurrentHashMap<>();
+
+    public static final TemporalAmountParser<Duration> JAIL_TIME_UNITS = new DurationParser()
+        .withUnit("s", ChronoUnit.SECONDS)
+        .withUnit("m", ChronoUnit.MINUTES)
+        .withUnit("h", ChronoUnit.HOURS)
+        .withUnit("d", ChronoUnit.DAYS);
+
+
+    @Inject
+    JailServiceImpl(TeleportService teleportService, SpawnService spawnService, NoticeService noticeService, PrisonersRepository prisonersRepository, JailLocationRepository jailLocationRepository, JailSettings settings) {
+        this.teleportService = teleportService;
+        this.spawnService = spawnService;
+        this.noticeService = noticeService;
+        this.settings = settings;
+        this.prisonersRepository = prisonersRepository;
+        this.jailRepository = jailLocationRepository;
+
+        this.prisonersRepository.getPrisoners().thenApply(prisoners -> {
+            for (Prisoner prisoner : prisoners) {
+                this.jailedPlayers.put(prisoner.getUuid(), prisoner);
+            }
+            return null;
+        });
+
+        this.jailRepository.getJailLocation().whenComplete((position, throwable) -> {
+            if (throwable != null) {
+                throwable.printStackTrace();
+                return;
+            }
+
+            position.ifPresent(value -> this.jailPosition = value);
+        });
+    }
+
+    @Override
+    public Map<UUID, Prisoner> getJailedPlayers() {
+        return this.jailedPlayers;
+    }
+
+    @Override
+    public Location getJailPosition() {
+        return PositionAdapter.convert(this.jailPosition);
+    }
+
+    @Override
+    public void setupJailArea(Location jailLocation, Player setter) {
+        if (this.isLocationSet()) {
+            this.noticeService.create()
+                .notice(translation -> translation.jailSection().jailLocationOverride())
+                .player(setter.getUniqueId())
+                .send();
+        }
+        else {
+            this.noticeService.create()
+                .notice(translation -> translation.jailSection().jailLocationSet())
+                .player(setter.getUniqueId())
+                .send();
+        }
```

Comment:
You can inline the code here 

```suggestion
            this.noticeService.create()
                .notice(translation -> (this.isLocationSet() ? translation.jailSection().jailLocationOverride() : translation.jailSection().jailLocationSet())
                .player(setter.getUniqueId())
                .send();
```